cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -747,7 +766,33 @@ predicate readStep(Node node1, Content c, Node node2) {
  * Holds if values stored inside content `c` are cleared at node `n`.
  */
 predicate clearsContent(Node n, Content c) {
-  none() // stub implementation
+  n =
+    any(PostUpdateNode pun, Content d | d.impliesClearOf(c) and storeStepImpl(_, d, pun, true) | pun)
+        .getPreUpdateNode() and
```

Comment:
Can we not remove this line now that we have a `forex` down at the bottom?

Reply:
The `forex` is in one side of an `or`, so I don't think so. Unless, I do not fully understand `forex` (which is entirely possible).

Reply:
We discussed this in a sync, and agreed that we cannot remove this line (for the exact reason that @jketema mentioned).

FWIW, `forex(a | b | c)` is functionality-wise identical to `exists(a | b | c) and forall(a | b | c)` (but may not always perform identically to the unfolded version).